feature-libs/checkout/base/core/facade/checkout.service.spec.ts

```diff
@@ -0,0 +1,128 @@
+import { AbstractType, Type } from '@angular/core';
+import { inject, TestBed } from '@angular/core/testing';
+import { MockStore, provideMockStore } from '@ngrx/store/testing';
+import { OrderPlacedEvent } from '@spartacus/checkout/base/root';
+import {
+  ActiveCartService,
+  Cart,
+  CartActions,
+  EventService,
+  OCC_USER_ID_CURRENT,
+  Order,
+  UserIdService,
+} from '@spartacus/core';
+import { Observable, of } from 'rxjs';
+import { CheckoutConnector } from '../connectors/checkout/checkout.connector';
+import { CheckoutService } from './checkout.service';
+
+const mockUserId = OCC_USER_ID_CURRENT;
+const mockCartId = 'cartID';
+const mockOrder: Order = { code: 'mockOrderCode' };
+
+class MockActiveCartService implements Partial<ActiveCartService> {
+  takeActiveCartId(): Observable<string> {
+    return of(mockCartId);
+  }
+  isGuestCart(_cart?: Cart): boolean {
+    return false;
+  }
+}
+
+class MockUserIdService implements Partial<UserIdService> {
+  takeUserId(_loggedIn = false): Observable<string> {
+    return of(mockUserId);
+  }
+}
+
+class MockCheckoutConnector implements Partial<CheckoutConnector> {
+  placeOrder(
+    _userId: string,
+    _cartId: string,
+    _termsChecked: boolean
+  ): Observable<Order> {
+    return of(mockOrder);
+  }
+}
+
+class MockEventService implements Partial<EventService> {
+  get<T>(_eventType: AbstractType<T>): Observable<T> {
+    return of();
+  }
+  dispatch<T extends object>(_event: T, _eventType?: Type<T>): void {}
+}
+
+describe(`CheckoutService`, () => {
+  let service: CheckoutService;
+  let connector: CheckoutConnector;
+  let eventService: EventService;
+  let store: MockStore;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        CheckoutService,
+        provideMockStore(),
+        { provide: ActiveCartService, useClass: MockActiveCartService },
+        { provide: UserIdService, useClass: MockUserIdService },
+        {
+          provide: CheckoutConnector,
+          useClass: MockCheckoutConnector,
+        },
+        { provide: EventService, useClass: MockEventService },
+      ],
+    });
+
+    service = TestBed.inject(CheckoutService);
+    connector = TestBed.inject(CheckoutConnector);
+    eventService = TestBed.inject(EventService);
+    store = TestBed.inject(MockStore);
+  });
+
+  it(`should inject CheckoutService`, inject(
+    [CheckoutService],
+    (checkoutService: CheckoutService) => {
+      expect(checkoutService).toBeTruthy();
+    }
+  ));
+
+  describe(`placeOrder`, () => {
+    const termsChecked = true;
+    it(`should call checkoutConnector.placeOrder`, () => {
+      spyOn(connector, 'placeOrder').and.callThrough();
+
+      service.placeOrder(termsChecked);
+
+      expect(connector.placeOrder).toHaveBeenCalledWith(
+        mockUserId,
+        mockCartId,
+        termsChecked
+      );
+    });
+
```

Comment:
reminder for me again:

- double check order (which is why it's not here), and is dependent on the answer